src/Compilers/CSharp/Portable/Binder/Binder_NameConflicts.cs

```diff
@@ -95,30 +90,48 @@ private static Location GetLocation(Symbol symbol)
         /// <remarks>
         /// Don't call this one directly - call one of the helpers.
         /// </remarks>
-        protected bool ValidateNameConflictsInScope(Symbol symbol, Location location, string name, DiagnosticBag diagnostics)
+        private bool ValidateNameConflictsInScope(Symbol symbol, Location location, string name, bool outsideContainingSymbol, DiagnosticBag diagnostics)
         {
             if (string.IsNullOrEmpty(name))
             {
                 return false;
             }
 
-            bool error = false;
+            // If the symbol is defined in a local function and shadowing is enabled,
+            // avoid checking for conflicts outside of the local function.
+            var stopAtLocalFunction =
+                symbol?.DeclaringCompilation.IsFeatureEnabled(MessageID.IDS_FeatureStaticLocalFunctions) == true ?
+                symbol.ContainingSymbol as LocalFunctionSymbol :
```

Comment:
>symbol.ContainingSymbol as LocalFunctionSymbol [](start = 16, length = 46)

Is this condition really necessary? For example, what if the symbol is a local or parameter in a lambda within a local function?

Reply:
Good catch, thanks.

---
In reply to: [245098932](https://github.com/dotnet/roslyn/pull/32067#discussion_r245098932) [](ancestors = 245098932)